### ISSUE TITLE
FakeTensor returns incorrect shapes for nansum and nanmean with empty dim

### ISSUE THREAD

**Issue by napronald** [pytorch/pytorch#191188]
### 🐛 Describe the bug

`torch.nansum` and `torch.nanmean` reduce over all dimensions when `dim=()` or `dim=[]` in eager mode, but FakeTensor incorrectly preserves the input shape. This causes valid programs to fail during `torch.compile` and `torch.export`.

## Reproduce

```python
import torch
from torch._subclasses.fake_tensor import FakeTensorMode

x = torch.randn(2, 3)

for op in (torch.nansum, torch.nanmean):
    for dim in ((), []):
        for keepdim in (False, True):
            eager = op(x, dim=dim, keepdim=keepdim)
            with FakeTensorMode() as mode:
                fake = op(mode.from_tensor(x), dim=dim, keepdim=keepdim)
            print(op.__name__, dim, keepdim, eager.shape, fake.shape)
```

Output

```text
nansum  () False torch.Size([])     torch.Size([2, 3])
nansum  () True  torch.Size([1, 1]) torch.Size([2, 3])
nansum  [] False torch.Size([])     torch.Size([2, 3])
nansum  [] True  torch.Size([1, 1]) torch.Size([2, 3])
nanmean () False torch.Size([])     torch.Size([2, 3])
nanmean () True  torch.Size([1, 1]) torch.Size([2, 3])
nanmean [] False torch.Size([])     torch.Size([2, 3])
nanmean [] True  torch.Size([1, 1]) torch.Size([2, 3])
```

The incorrect metadata also breaks graph capture:

```python
def fn(x):
    return torch.nansum(x, dim=()).view(())

torch.compile(fn, backend="eager", fullgraph=True)(x)
```

Eager execution succeeds, but compilation fails during FakeTensor execution.

### Versions

```text
PyTorch: 2.14.0.dev20260726
Python: 3.12
```

cc @chauhang @penguinwu @eellison @aorenste @liangel-02 @bdhirsh @bobrenjc93

**Comment by napronald** [comment]
I'd be happy to work on this. Could it be assigned to me?
